cli/azd/internal/agent/logging/chained_handler.go

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package logging
+
+import (
+	"context"
+
+	"github.com/tmc/langchaingo/callbacks"
+	"github.com/tmc/langchaingo/llms"
+	"github.com/tmc/langchaingo/schema"
+)
+
+// ChainedHandler forwards calls to multiple callbacks.Handler in order.
+type ChainedHandler struct {
+	handlers []callbacks.Handler
+}
+
+// NewChainedHandler creates a new ChainedHandler with the provided handlers.
+func NewChainedHandler(handlers ...callbacks.Handler) *ChainedHandler {
+	return &ChainedHandler{handlers: handlers}
+}
+
+// Compile-time check to ensure ChainedHandler implements callbacks.Handler
+var _ callbacks.Handler = &ChainedHandler{}
```

Comment:
Same as `ActionLogger`, the `ChainedHandler` should define the intention to implement `callbacks.Handler` in the right way.
Also, would this be `ChainedLogger` instead of `ChainedHandler` ?

Reply:
It could be used to create any chained callback handler - not just for logging.

Reply:
Removing naming-suggestions as a `nit`.